contracts/dev-utils/contracts/src/LibDydxBalance.sol

```diff
@@ -0,0 +1,418 @@
+                info,
+                withdraw
+            );
+            // If the deposit to withdraw ratio is >= the minimum collateralization
+            // ratio, then we will never become insolvent at these prices.
+            if (D18.div(dd, db) >= minCr) {
+                continue;
+            }
+            // The collateralization ratio for this account, parameterized by
+            // `t` (maker amount), is given by:
+            //      `cr = (supplyValue + t * dd) / (borrowValue + t * db)`
+            // Solving for `t` gives us:
+            //      `t = (supplyValue - cr * borrowValue) / (cr * db - dd)`
+            int256 t = D18.div(
+                D18.sub(supplyValue, D18.mul(minCr, borrowValue)),
+                D18.sub(D18.mul(minCr, db), dd)
+            );
+            solventMakerAmount = LibSafeMath.min256(
+                solventMakerAmount,
+                // `t` is in maker token units, so convert it to maker wei.
+                _toWei(info.makerTokenAddress, uint256(D18.clip(t)))
+            );
+        }
+    }
+
+    /// @dev Create a `BalanceCheckInfo` struct.
+    /// @param order An order with a `DydxBridge` maker asset.
+    /// @param dydx The address of the Dydx contract.
+    /// @return info The `BalanceCheckInfo` struct.
+    function _getBalanceCheckInfo(LibOrder.Order memory order, address dydx)
+        private
+        pure
+        returns (BalanceCheckInfo memory info)
+    {
+        bytes memory rawBridgeData;
+        (, info.makerTokenAddress, info.bridgeAddress, rawBridgeData) =
+            LibAssetData.decodeERC20BridgeAssetData(order.makerAssetData);
+        info.dydx = IDydx(dydx);
+        info.makerAddress = order.makerAddress;
+        if (order.takerAssetData.readBytes4(0) == IAssetData(0).ERC20Token.selector) {
+            (, info.takerTokenAddress) =
+                LibAssetData.decodeERC20AssetData(order.takerAssetData);
+        }
+        info.orderMakerToTakerRate = D18.div(order.takerAssetAmount, order.makerAssetAmount);
+        (IDydxBridge.BridgeData memory bridgeData) =
+            abi.decode(rawBridgeData, (IDydxBridge.BridgeData));
+        info.accounts = bridgeData.accountNumbers;
+        info.actions = bridgeData.actions;
+    }
+
+    /// @dev Returns the conversion rate for an action.
+    /// @param action A `BridgeAction`.
+    function _getActionRate(IDydxBridge.BridgeAction memory action)
+        private
+        pure
+        returns (int256 rate)
+    {
+        rate = action.conversionRateDenominator == 0
+            ? D18.one()
+            : D18.div(
+                action.conversionRateNumerator,
+                action.conversionRateDenominator
+            );
+    }
+
+    /// @dev Returns the USD value of an action based on its conversion rate
+    ///      and market prices.
+    /// @param info State from `_getBalanceCheckInfo()`.
+    /// @param action A `BridgeAction`.
+    function _getActionRateValue(
```

Comment:
Nit - I think a name like `_getMonetaryValue` could be more descriptive and aligned with naming used by dYdX.